testsuite/forge/src/tx_emitter.rs

```diff
@@ -0,0 +1,717 @@
+        let mut rng = ThreadRng::default();
+        let batch_size = max(MAX_TXN_BATCH_SIZE, self.accounts.len());
+        let accounts = self
+            .accounts
+            .iter_mut()
+            .choose_multiple(&mut rng, batch_size);
+        let mut requests = Vec::with_capacity(accounts.len());
+        for sender in accounts {
+            let receiver = self
+                .all_addresses
+                .choose(&mut rng)
+                .expect("all_addresses can't be empty");
+            let request =
+                gen_transfer_txn_request(sender, receiver, SEND_AMOUNT, self.tx_factory.clone());
+            requests.push(request);
+        }
+        requests
+    }
+}
+
+#[derive(Debug)]
+pub struct TxEmitter<'t> {
+    accounts: Vec<LocalAccount>,
+    tx_factory: TransactionFactory,
+    network_info: NetworkInfo<'t>,
+    client: JsonRpcClient,
+}
+
+impl<'t> TxEmitter<'t> {
+    pub fn new(network_info: NetworkInfo<'t>) -> Self {
+        let tx_factory = TransactionFactory::new(network_info.chain_id());
+        let client = JsonRpcClient::new(network_info.json_rpc());
+        Self {
+            accounts: vec![],
+            tx_factory,
+            network_info,
+            client,
+        }
+    }
+
+    pub async fn get_money_source(&mut self, coins_total: u64) -> Result<&mut LocalAccount> {
+        let mut client = self.client.clone();
+        println!("Creating and minting faucet account");
+        let mut faucet_account = &mut self.network_info.dd_account;
+        let mint_txn = gen_transfer_txn_request(
+            faucet_account,
+            &faucet_account.address(),
+            coins_total,
+            self.tx_factory.clone(),
+        );
+        execute_and_wait_transactions(&mut client, &mut faucet_account, vec![mint_txn])
+            .await
+            .map_err(|e| format_err!("Failed to mint into faucet account: {}", e))?;
+        let balance = retrieve_account_balance(&client, faucet_account.address()).await?;
+        for b in balance {
+            if b.currency.eq(XUS_NAME) {
+                println!(
+                    "DD account current balances are {}, requested {} coins",
+                    b.amount, coins_total
+                );
+                break;
+            }
+        }
+        Ok(faucet_account)
+    }
+
+    pub async fn get_seed_accounts(
+        &mut self,
+        json_rpc_clients: &[JsonRpcClient],
+        seed_account_num: usize,
+    ) -> Result<Vec<LocalAccount>> {
+        info!("Creating and minting seeds accounts");
+        let tx_factory = self.tx_factory.clone();
+        let mut i = 0;
+        let mut seed_accounts = vec![];
+        while i < seed_account_num {
+            let mut client = self.pick_mint_client(json_rpc_clients).clone();
+            let mut creation_account = &mut self.network_info.treasury_compliance_account;
+            let mut batch = gen_random_accounts(min(MAX_TXN_BATCH_SIZE, seed_account_num - i));
```

Comment:
Might follow a little easier if you pull out the `min` statement, and call it batch size.  Then add that to `i`.